### PR TITLE
IPv6 communication over VPC peering

### PR DESCRIPTION
**Pull request by ngoyal16**
## what
* Allow IPv6 communication b/w VPCs over VPC peering

## why
* Application hosted in VPC-1 wants to access resources hosted in private subnet of VPC-2 over IPv6.

### CODE REVIEW

File: requester.tf [hunk 1]
```diff
@@ -160,19 +160,28 @@ resource "aws_vpc_peering_connection_options" "requester" {
...
+  requester_aws_route_table_ids_count = length(local.requester_aws_route_table_ids)
+  requester_ipv6_cidr_blocks = flatten(length(data.aws_vpc.requester.*.ipv6_cidr_block) > 0 ? [
+    for vpc_temp in data.aws_vpc.requester : {
+      cidr_block = vpc_temp.ipv6_cidr_block
+    }
+  ] : [])
+  requester_cidr_block_associations = flatten([
+    data.aws_vpc.requester.*.cidr_block_associations,
+    local.requester_ipv6_cidr_blocks
+  ])
```
Comment (Nuru) [inline requester.tf:173]: Same as with "accepter": Why is this change necessary? `cidr_block_associations` should have all the IPv4 and IPv6 CIDRs.

File: requester.tf [hunk 2]
```diff
@@ -160,19 +160,28 @@ resource "aws_vpc_peering_connection_options" "requester" {
...
 resource "aws_route" "requester" {
-  count                     = local.enabled ? local.requester_aws_route_table_ids_count * local.accepter_cidr_block_associations_count : 0
-  provider                  = aws.requester
-  route_table_id            = local.requester_aws_route_table_ids[floor(count.index / local.accepter_cidr_block_associations_count)]
-  destination_cidr_block    = local.accepter_cidr_block_associations[count.index % local.accepter_cidr_block_associations_count]["cidr_block"]
-  vpc_peering_connection_id = join("", aws_vpc_peering_connection.requester.*.id)
+  count                       = local.enabled ? local.requester_aws_route_table_ids_count * local.accepter_cidr_block_associations_count : 0
+  provider                    = aws.requester
+  route_table_id              = local.requester_aws_route_table_ids[floor(count.index / local.accepter_cidr_block_associations_count)]
+  destination_cidr_block      = can(regex("::", local.accepter_cidr_block_associations[count.index % local.accepter_cidr_block_associations_count]["cidr_block"])) ? null : local.accepter_cidr_block_associations[count.index % local.accepter_cidr_block_associations_count]["cidr_block"]
```
Comment (Nuru) [inline requester.tf:182]: I hate using `regex` here,  but could live with it. I definitely do not want to use `can()` here. We don't need a regex, anyway. Use

```hcl
split(":", local.accepter_cidr_block_associations[count.index % local.accepter_cidr_block_associations_count]["cidr_block"]) > 1
```
to determine that the string is an IPv6 CIDR

**Review by Nuru — CHANGES_REQUESTED** [review]

File: accepter.tf [hunk 3]
```diff
@@ -88,21 +88,30 @@ data "aws_route_tables" "default_rts" {
...
+  accepter_aws_route_table_ids_count = length(local.accepter_aws_route_table_ids)
+  accepter_ipv6_cidr_blocks = flatten(length(data.aws_vpc.accepter.*.ipv6_cidr_block) > 0 ? [
+    for vpc_temp in data.aws_vpc.accepter : {
+      cidr_block = vpc_temp.ipv6_cidr_block
+    }
+  ] : [])
+  accepter_cidr_block_associations = flatten([
+    data.aws_vpc.accepter.*.cidr_block_associations,
+    local.accepter_ipv6_cidr_blocks
+  ])
```
[the remark this answers is not in the document]
Comment (ngoyal16) [reply accepter.tf:103]: @Nuru  `cidr_block_associations` do not contains IPv6 CIDRs

File: requester.tf (hunk 1 above)
Comment (ngoyal16) [reply requester.tf:173]: @Nuru  `cidr_block_associations` do not contains IPv6 CIDRs

File: requester.tf (hunk 2 above)
Comment (ngoyal16) [reply requester.tf]: @Nuru  PR has been updated. can you check and let me know if anything extra needs to be added.

**Pull request closed without merging**